main.go

```diff
@@ -79,10 +80,24 @@ func main() {
 		os.Exit(1)
 	}
 
+	ctx := ctrl.SetupSignalHandler()
+
+	kubeConfig, err := ctrl.GetConfig()
+	if err != nil {
+		setupLog.Error(err, "unable to get kubeConfig", "controller", "SearchOperator")
+		os.Exit(1)
+	}
+	addonMgr, err := addon.NewAddonManager(kubeConfig)
+	if err != nil {
+		setupLog.Error(err, "unable to create a new  addon manager", "controller", "SearchOperator")
```

Comment:
Is it possible to recover from this error? I wonder if there's any benefit from adding a retry logic.

Reply:
I don't think so. `addonMgr` is a controller,  and `addon.NewAddonManager` only does some static logics.  I think it is impossible to recover if there is error.
so could we consider to` os.Exit(1)` here ?  because search-collector-addon as a part of search-operator, the whole search feature cannot work if the addonMgr does not start working.

Reply:
If we hit this error, search could still work, but it will be degraded to only resources in the hub. If we exit, then the hub components won't get deployed, which causes a worse problem.

Both scenarios aren't good, but exiting will cause a more severe problem. On the other hand, continuing could make it harder to notice and debug the problem. Could we show this error in the operator's status?

@xrajesh What are your thoughts for this?

Reply:
@zhiweiyin318 We can put a log message when the addon is successfully created - Helps if there is a problem, because we are quite when everything is working .